parlai/core/utils.py

```diff
@@ -311,7 +311,13 @@ def round_sigfigs(x, sigfigs=4):
 
 
 def flatten(teacher, context_length=-1, include_labels=True):
-    """Return a flattened version of a teacher's data.
+    """
+    ******************************************************************
+    DEPRECATED: If you would like to make use of batch sorting, please
+    use the PytorchDataTeacher
+    ******************************************************************
+
+    Return a flattened version of a teacher's data.
```

Comment:
Humble recommendation: add

warnings.warn("flatten is deprecated. Please use PytorchDataTeacher.", warnings.DeprecationWarning)

Reply:
Also make sure this renders reasonably in the autodocs.

Reply:
Thanks; will make sure the autodocs look good.